libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/MapboxRouteOptionsUpdater.kt

```diff
@@ -146,13 +146,13 @@ class MapboxRouteOptionsUpdater(
             if (legacyBearingList != null) {
                 newList.addAll(
                     legacyBearingList.subList(
-                        legIndex + 1,
+                        coordinates - remainingWaypoints,
```

Comment:
@Lebedinsky can you add/modify a test in `MapboxRouteOptionsUpdaterTest`? it should fail before this change, and pass after this change.